From c1af14126dd05300fddc6c26d6d59673606c9d69 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Fri, 21 Feb 2014 12:42:26 +0100
Subject: [PATCH 5/7] arch-zynq/hardware.h: Fix bootmode mask

Bootmode mask was defined as 0x0F, but documentation mentions 0x07.
Experiments show that bit "3" is the JTAG chain configuration.
Change the mask to "7" to allow systems with a different chain
configuration to boot correctly.

Signed-off-by: Mike Looijmans <mike.looijmans@topic.nl>
---
 arch/arm/include/asm/arch-zynq/hardware.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/asm/arch-zynq/hardware.h b/arch/arm/include/asm/arch-zynq/hardware.h
index f126179..4925fac 100644
--- a/arch/arm/include/asm/arch-zynq/hardware.h
+++ b/arch/arm/include/asm/arch-zynq/hardware.h
@@ -28,7 +28,7 @@
 #define ZYNQ_EFUSE_BASEADDR		0xF800D000
 
 /* Bootmode setting values */
-#define ZYNQ_BM_MASK		0x0F
+#define ZYNQ_BM_MASK		0x07
 #define ZYNQ_BM_QSPI		0x01
 #define ZYNQ_BM_NOR		0x02
 #define ZYNQ_BM_NAND		0x04
-- 
1.7.9.5

